src/DynamoCore/Configuration/PreferenceSettings.cs

```diff
@@ -288,6 +288,13 @@ public int MaxNumRecentFiles
         /// </summary>
         public List<string> CustomPackageFolders { get; set; }
 
+        /// <summary>
+        /// A list of trusted locations (folders) as it is recorded in the preferences file.
+        /// Do not directly use this property when trying to check for trusted locations.
+        /// Instead use Dynamo.Core.TrustedLocatationsManager to check/query for trusted locations.
+        /// </summary>
+        public List<string> TrustedLocations { get; set; }
```

Comment:
can you make the setter internal / private so that extensions cannot easily modify this?